### ISSUE TITLE
💅 `useAwaitThenable` incorrectly inferring type of `super.fn()`

### ISSUE THREAD

**Issue by DayKev** [biomejs/biome#8481]
### Environment information

```bash
CLI:
  Version:                      2.3.9
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.25.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    nursery/useAwaitThenable

Workspace:
  Open Documents:               0
```

### Rule name

lint/nursery/useAwaitThenable

### Playground link

https://github.com/DayKev/use-await-thenable

### Expected result

There should be no diagnostic on the [`files = await super.sort(files)`](https://github.com/DayKev/use-await-thenable/blob/3c2f8e32fe05cb655e1f08b62ba266e54f50830f/src/index.ts#L6) line due to that method being `async`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
cc @arendjr

**Comment by ematipico** [comment]
Closing as it's been fixed. The playground doesn't show any diagnostic https://biomejs.dev/playground/?lintRules=useAwaitThenable&problems=console#code=aQBtAHAAbwByAHQAIAB7ACAAQgBhAHMAZQBTAGUAcQB1AGUAbgBjAGUAcgAsACAAdAB5AHAAZQAgAFQAZQBzAHQAUwBwAGUAYwBpAGYAaQBjAGEAdABpAG8AbgAgAH0AIABmAHIAbwBtACAAIgB2AGkAdABlAHMAdAAvAG4AbwBkAGUAIgA7AAoACgBlAHgAcABvAHIAdAAgAGMAbABhAHMAcwAgAE0AeQBTAGUAcQB1AGUAbgBjAGUAcgAgAGUAeAB0AGUAbgBkAHMAIABCAGEAcwBlAFMAZQBxAHUAZQBuAGMAZQByACAAewAKAAkALwAvACAAYwBmACAAaAB0AHQAcABzADoALwAvAGcAaQB0AGgAdQBiAC4AYwBvAG0ALwB2AGkAdABlAHMAdAAtAGQAZQB2AC8AdgBpAHQAZQBzAHQALwBiAGwAbwBiAC8ANgA0ADkANwBiAGIAMgA2ADAAOABmADUANAA0AGUAMwA4AGYAZABiAGIAOQAxAGQAYQA1AGQAOQAzADMAMQA0AGEAZAAwADEANQA4ADMAYgAvAHAAYQBjAGsAYQBnAGUAcwAvAHYAaQB0AGUAcwB0AC8AcwByAGMALwBuAG8AZABlAC8AcwBlAHEAdQBlAG4AYwBlAHIAcwAvAEIAYQBzAGUAUwBlAHEAdQBlAG4AYwBlAHIALgB0AHMAIwBMADMANQAKAAkAbwB2AGUAcgByAGkAZABlACAAYQBzAHkAbgBjACAAcwBvAHIAdAAoAGYAaQBsAGUAcwA6ACAAVABlAHMAdABTAHAAZQBjAGkAZgBpAGMAYQB0AGkAbwBuAFsAXQApADoAIABQAHIAbwBtAGkAcwBlADwAVABlAHMAdABTAHAAZQBjAGkAZgBpAGMAYQB0AGkAbwBuAFsAXQA%2BACAAewAKAAkACQBmAGkAbABlAHMAIAA9ACAAYQB3AGEAaQB0ACAAcwB1AHAAZQByAC4AcwBvAHIAdAAoAGYAaQBsAGUAcwApADsAIAAvAC8AIABpAG4AZgBlAHIAcgBpAG4AZwAgAHQAaABlACAAdwByAG8AbgBnACAAdAB5AHAAZQAgAGYAbwByACAAYABzAHUAcABlAHIALgBzAG8AcgB0AGAAPwAKAAoACQAJAHIAZQB0AHUAcgBuACAAcwB1AHAAZQByAC4AcwBvAHIAdAAoAGYAaQBsAGUAcwApADsAIAAvAC8AIAB1AG4AcgBlAGwAYQB0AGUAZAAgAGMAbwBkAGUAIAByAGUAbQBvAHYAZQBkAAoACQB9AAoAfQAKAAoAYQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAZgBuACgAKQA6ACAAUAByAG8AbQBpAHMAZQA8AFQAZQBzAHQAUwBwAGUAYwBpAGYAaQBjAGEAdABpAG8AbgBbAF0APgAgAHsACgAJAHIAZQB0AHUAcgBuACAAWwBdADsACgB9AAoACgBhAHcAYQBpAHQAIABmAG4AKAApADsAIAAvAC8AIABuAG8AIABlAHIAcgBvAHIALAAgAHMAYQBtAGUAIAB0AHkAcABlACAAcwBpAGcAbgBhAHQAdQByAGUACgA%3D
